test/foundry/mocks/hts-precompile/HederaNonFungibleToken.sol

```diff
@@ -0,0 +1,179 @@
+    constructor(
+        IHederaTokenService.TokenInfo memory _nftTokenInfo
+    ) ERC721(_nftTokenInfo.token.name, _nftTokenInfo.token.symbol) {
+        address sender = msg.sender;
+        HtsPrecompile.registerHederaNonFungibleToken(sender, _nftTokenInfo);
+    }
+
+    /// @dev the HtsSystemContractMock should do precheck validation before calling any function with this modifier
+    ///      the HtsSystemContractMock has priveleged access to do certain operations
+    modifier onlyHtsPrecompile() {
+        require(msg.sender == HTS_PRECOMPILE, 'NOT_HTS_PRECOMPILE');
+        _;
+    }
+
+    // public/external state-changing functions:
+    // onlyHtsPrecompile functions:
+    function mintRequestFromHtsPrecompile(
+        bytes[] memory metadata
+    ) external onlyHtsPrecompile returns (int64 newTotalSupply, int64 serialNumber) {
+        (, IHederaTokenService.TokenInfo memory nftTokenInfo) = HtsPrecompile.getTokenInfo(
+            address(this)
+        );
+        address treasury = nftTokenInfo.token.treasury;
+
+        serialNumber = ++nftCount.minted; // the first nft that is minted has serialNumber: 1
+        _mint(treasury, uint64(serialNumber));
+
+        newTotalSupply = int64(int256(totalSupply()));
+    }
+
+    function burnRequestFromHtsPrecompile(
+        int64[] calldata tokenIds
+    ) public onlyHtsPrecompile returns (int64 newTotalSupply) {
+        int64 burnCount = int64(uint64(tokenIds.length));
+        nftCount.burned = nftCount.burned + burnCount;
+
+        for (uint256 i = 0; i < uint64(burnCount); i++) {
+            uint256 tokenId = uint64(tokenIds[i]);
+            _burn(tokenId);
+        }
+
+        newTotalSupply = int64(int256(totalSupply()));
+    }
+
+    function wipeRequestFromHtsPrecompile(
+        int64[] calldata tokenIds
+    ) external onlyHtsPrecompile {
+        burnRequestFromHtsPrecompile(tokenIds); // implementation happens to coincide with burnRequestFromHtsPrecompile unlike in HederaFungibleToken
+    }
+
+    /// @dev transfers "amount" from "from" to "to"
+    function transferRequestFromHtsPrecompile(
+        bool isRequestFromOwner,
+        address spender,
+        address from,
+        address to,
+        uint256 tokenId
+    ) external onlyHtsPrecompile returns (int64 responseCode) {
+        bool isSpenderApproved = _isApprovedOrOwner(spender, tokenId);
+        if (!isSpenderApproved) {
+            return HederaResponseCodes.INSUFFICIENT_TOKEN_BALANCE;
+        }
+
+        _transfer(from, to, tokenId);
+        responseCode = HederaResponseCodes.SUCCESS;
+        // if (getApproved(tokenId) == spender) {
+        //     _transfer(from, to, tokenId);
+        //     responseCode = HederaResponseCodes.SUCCESS;
+        // } else {
+        //     responseCode = HederaResponseCodes.INSUFFICIENT_TOKEN_BALANCE;
+        // }
```

Comment:
Why this commented code? is it intentional?

Reply:
It can be removed the validation is/should be done in the `HtsSystemContractMock` contract